libraries/AP_Landing/AP_Landing_Deepstall.cpp

```diff
@@ -0,0 +1,380 @@
+        if (get_distance(current_loc, type_deepstall_landing_point) > 2 * aparm.loiter_radius) {
+            nav_controller->update_waypoint(current_loc, type_deepstall_landing_point);
+            return false;
+        }
+        type_deepstall_stage = DEEPSTALL_STAGE_ESTIMATE_WIND;
+        type_deepstall_loiter_sum_cd = 0; // reset the loiter counter
+        //fallthrough
+    case DEEPSTALL_STAGE_ESTIMATE_WIND:
+        {
+        if (!nav_controller->reached_loiter_target() || (fabsf(height) > DEEPSTALL_LOITER_ALT_TOLERANCE)) {
+            // wait until the altitude is correct before considering a breakout
+            nav_controller->update_loiter(type_deepstall_landing_point, aparm.loiter_radius, 1);
+            return false;
+        }
+        // only count loiter progress when within the target altitude
+        int32_t target_bearing = nav_controller->target_bearing_cd();
+        int32_t delta = wrap_180_cd(target_bearing - type_deepstall_last_target_bearing);
+        if (delta > 0) { // only accumulate turns in the correct direction
+            type_deepstall_loiter_sum_cd += delta;
+        }
+        type_deepstall_last_target_bearing = target_bearing;
+        if (type_deepstall_loiter_sum_cd < 36000) {
+            // wait until we've done at least one complete loiter at the correct altitude
+            nav_controller->update_loiter(type_deepstall_landing_point, aparm.loiter_radius, 1);
+            return false;
+        }
+        type_deepstall_stage = DEEPSTALL_STAGE_WAIT_FOR_BREAKOUT;
+        //compute optimal path for landing
+        type_deepstall_build_approach_path();
+        //fallthrough
+        }
+    case DEEPSTALL_STAGE_WAIT_FOR_BREAKOUT:
+        if (!type_deepstall_verify_breakout(current_loc, type_deepstall_arc_entry, height)) {
+            nav_controller->update_loiter(type_deepstall_landing_point, aparm.loiter_radius, 1);
+            return false;
+        }
+        type_deepstall_stage = DEEPSTALL_STAGE_FLY_TO_ARC;
+        memcpy(&type_deepstall_breakout_location, &current_loc, sizeof(Location));
+        //fallthrough
+    case DEEPSTALL_STAGE_FLY_TO_ARC:
+        if (get_distance(current_loc, type_deepstall_arc_entry) > 2 * aparm.loiter_radius) {
+            nav_controller->update_waypoint(type_deepstall_breakout_location, type_deepstall_arc_entry);
+            return false;
+        }
+        type_deepstall_stage = DEEPSTALL_STAGE_ARC;
+        //fallthrough
+    case DEEPSTALL_STAGE_ARC:
+        {
+        Vector2f groundspeed = ahrs.groundspeed_vector();
+        if (!nav_controller->reached_loiter_target() ||
+            (fabsf(wrap_180(type_deepstall_target_heading_deg -
+                            degrees(atan2f(-groundspeed.y, -groundspeed.x) + M_PI))) >= 10.0f)) {
+            nav_controller->update_loiter(type_deepstall_arc, aparm.loiter_radius, 1);
+            return false;
+        }
+        type_deepstall_stage = DEEPSTALL_STAGE_APPROACH;
+        //fallthrough
```

Comment:
Everything within the case is inside the {}'s nothing is meant to be outside of it. I can move it down I guess, but that mentally broke the scoping in my head...